test/parallel/test-http-socket-listeners.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+const net = require('net');
+
+const server = http.createServer((req, res) => {
+  res.writeHead(200);
+  res.end('response');
+});
+
+server.on('clientError', common.mustCallAtLeast(() => {}, 1));
```

Comment:
This is invalid API usage. The documentation says that "Listener of this event is responsible for closing/destroying the underlying socket."

Reply:
I agree. But if the user misbehave or the server is hammered with bad data before the user can handle the socket (like it seems to happen in the issue referenced by this PR) then we should make sure we don't leak listeners.